server/src/main/java/org/opensearch/indices/recovery/RecoverySettings.java

```diff
@@ -157,6 +157,17 @@ public class RecoverySettings {
         Property.NodeScope
     );
 
+    /**
+     * Controls minimum number of metadata files to keep in remote segment store.
+     * {@code value < 1} will disable deletion of stale segment metadata files.
+     */
+    public static final Setting<Integer> CLUSTER_REMOTE_INDEX_MIN_SEGMENT_METADATA_FILES_SETTING = Setting.intSetting(
```

Comment:
Lets also create the issue for the target state fix (lease based deletion of segments/translog files on remote store). What is the plan around deprecating this setting once we have lease based deletion for segments and translog files stored on remote store? If we plan to keep this, what will this setting be used for?

Reply:
https://github.com/opensearch-project/OpenSearch/issues/11264

Reply:
> What is the plan around deprecating this setting once we have lease based deletion for segments and translog files stored on remote store? If we plan to keep this, what will this setting be used for?

Once we have the long term fix around the segment metadata files (based on lease/locks), users who do not want to keep extra data in remote can change the value of this setting to `1`.  So, both the solutions can co-exist. We can take a call on deprecating this setting if don't see any further use cases.